sdk/core/dma/dma_compartment.cc

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <utils.hh>
+#include <thread_pool.h>
+
+#if __has_include(<platform-dma.hh>)
+#	include <platform-dma.hh>
+#endif
+
+// Import some useful things from the CHERI namespace.
+using namespace CHERI;
+
+using thread_pool::async;
+/**
+ *  creating a global heap below.
+ *  todo: we assume that this heap capability can serve 
```

Comment:
Yes, exactly. This bounds claims for the duration of a DMA and so has a large quota but one that does not contribute much to total memory consumption.